File: leetcode/Python/shift-2d-grid.py
class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        v = deepcopy(grid)
        m, n = len(grid), len(grid[0])
        ny, nx = 0, 0
        for i in range(m):
            for j in range(n):
                nx = j + k
                ny = i + nx//n
                ny, nx = ny%m, nx%n
                v[ny][nx]=grid[i][j]
        return v

File: leetcode/C++/shift-2d-grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        vector<vector<int>> v = grid;
        int m=grid.size(), n=grid[0].size();
        int ny=0,nx=0;
        for(int i=0; i<m; ++i){
            for(int j=0; j<n; ++j){
                nx=j+k; ny=i+nx/n;
                nx%=n; ny%=m;
                v[ny][nx]=grid[i][j];
            }
        }
        return v;
    }
};